pandas/tests/io/test_sql.py

```diff
@@ -1906,1102 +1924,1356 @@ def test_api_read_sql_duplicate_columns(conn, request):
-            Integer,
-            Unicode,
-            select,
+@pytest.mark.db
+@pytest.mark.parametrize("conn", all_connectable)
+@pytest.mark.parametrize("func", ["read_sql", "read_sql_query"])
+def test_read_sql_dtype_backend(
+    conn,
+    request,
+    string_storage,
+    func,
+    dtype_backend,
+    dtype_backend_data,
+    dtype_backend_expected,
+):
+    # GH#50048
+    conn_name = conn
+    conn = request.getfixturevalue(conn)
+    table = "test"
+    df = dtype_backend_data
+    df.to_sql(name=table, con=conn, index=False, if_exists="replace")
+
+    with pd.option_context("mode.string_storage", string_storage):
+        result = getattr(pd, func)(
+            f"Select * from {table}", conn, dtype_backend=dtype_backend
         )
-        from sqlalchemy.orm import (
-            Session,
-            declarative_base,
+    expected = dtype_backend_expected(string_storage, dtype_backend, conn_name)
+    tm.assert_frame_equal(result, expected)
+
+    with pd.option_context("mode.string_storage", string_storage):
+        iterator = getattr(pd, func)(
+            f"Select * from {table}",
+            con=conn,
+            dtype_backend=dtype_backend,
+            chunksize=3,
         )
+        expected = dtype_backend_expected(string_storage, dtype_backend, conn_name)
+        for result in iterator:
+            tm.assert_frame_equal(result, expected)
 
-        test_data = "Hello, World!"
-        expected = DataFrame({"spam": [test_data]})
-        Base = declarative_base()
-
-        class Temporary(Base):
-            __tablename__ = "temp_test"
-            __table_args__ = {"prefixes": ["TEMPORARY"]}
-            id = Column(Integer, primary_key=True)
-            spam = Column(Unicode(30), nullable=False)
-
-        with Session(self.conn) as session:
-            with session.begin():
-                conn = session.connection()
-                Temporary.__table__.create(conn)
-                session.add(Temporary(spam=test_data))
-                session.flush()
-                df = sql.read_sql_query(sql=select(Temporary.spam), con=conn)
-        tm.assert_frame_equal(df, expected)
-
-    # -- SQL Engine tests (in the base class for now)
-    def test_invalid_engine(self, test_frame1):
-        msg = "engine must be one of 'auto', 'sqlalchemy'"
-        with pytest.raises(ValueError, match=msg):
-            self._to_sql_with_sql_engine(test_frame1, "bad_engine")
 
-    def test_options_sqlalchemy(self, test_frame1):
-        # use the set option
-        with pd.option_context("io.sql.engine", "sqlalchemy"):
-            self._to_sql_with_sql_engine(test_frame1)
+@pytest.mark.db
+@pytest.mark.parametrize("conn", all_connectable)
+@pytest.mark.parametrize("func", ["read_sql", "read_sql_table"])
+def test_read_sql_dtype_backend_table(
+    conn,
+    request,
+    string_storage,
+    func,
+    dtype_backend,
+    dtype_backend_data,
+    dtype_backend_expected,
+):
+    if "sqlite" in conn:
```

Comment:
This is silently passed on main. What's interesting is that `read_sql_table` arguably works better than `read_sql` because it correctly maps the boolean columns to a boolean value. We can update the fixture to account for that, but didn't want to tackle here to try and reduce the already enormous diff